src/SpinWaveTheory/SWTCalculations.jl

```diff
@@ -3,7 +3,10 @@
 ###########################################################################
 
 @inline δ(x, y) = ==(x, y) # my delta function
-
+# The "metric" of scalar biquad interaction. Here we are using the following identity:
+# (𝐒ᵢ⋅𝐒ⱼ)² = -(𝐒ᵢ⋅𝐒ⱼ)/2 + ∑ₐ (OᵢᵃOⱼᵃ)/2, a=4,…,8, 
+# where the definition of Oᵢᵃ is given in Appendix B of *Phys. Rev. B 104, 104409*
+const biquad_metric = 1/2 * diagm([-1, -1, -1, 1, 1, 1, 1, 1])
 
 """
     generate_ham_lswt!
```

Comment:
I wonder if this function could be broken down a little bit. For example, there could be functions`set_swt_pair_exchange!`, `set_swt_quadratic_exchange!`, `set_swt_sun_aniso`, `set_swt_dipole_aniso` or similar.